content/blog/2023/2023-04-23-Java-9-module-system.md

```diff
@@ -0,0 +1,169 @@
+# Java 9 Module System
+
+Table of Contents
+
+1. [Introduction](#introduction)
+
+2. [Issues with Previous Java Versions](#issues-with-previous-java-versions)
+
+3. [Benefits of Java 9 Module System](#benefits-of-java-9-module-system)
+
+4. [What is a Module?](#what-is-a-module)
+
+5. [Creating the Modules](#creating-the-modules)
+
+6. [Structuring Our Project](#structuring-our-project)
+
+7. [Creating the First Module](#creating-the-first-module)
+8. [Creating the second module](#creating-the-second-module)
+9. [Compiling our modules](#compiling-our-modules)
+10. [Running our Modules](#running-our-modules)
+11. [Updated Module Descriptor](#updated-module-descriptor)
+12. [Conclusion](#conclusion)
+
+Java 9 introduced one of the most important features called a module system that allows us to combine all of our code and packages into a single unit. In this tutorial, we’ll discuss what is this module system. why should we use it and how should we use it? We’ll be looking at the answers to all of these questions. Also, we’ll look at a practical example to add glitters to our understanding, so let’s begin.
+
+## Introduction <a id='introduction'></a>
```

Comment:
No `<a>` tags necessary in headers, the blog does that automatically. Please remove.